src/components/NavItem/index.tsx

```diff
@@ -43,31 +52,85 @@ function _NavItem({
   collapseClass[styles.collapsed] = !!collapsed
   const Element = component || 'button'
   const linkClasses: Record<string, boolean> = {}
-  linkClasses[styles.selected] = currentRoute === route
+  linkClasses[styles.selected] = !!selected
 
   const linkItemClasses: Record<string, boolean> = {}
   linkItemClasses[styles.hide] = alertText === ''
   linkItemClasses[styles.secondary] = !!secondaryTag
+
+  let dropdownArrow = null
+  const hasSublinks = subLinks && subLinks.length > 0
+  if (hasSublinks) {
+    const dropdownArrowClass: Record<string, boolean> = {}
+    dropdownArrowClass[styles.subMenuCollapsed] = subLinksCollapsed
+    dropdownArrow = (
+      <button
+        className={cn(styles.submenuCollapseIcon, dropdownArrowClass)}
+        onClick={(ev) => {
+          setSubLinksCollapsed?.(!subLinksCollapsed)
+          ev.preventDefault()
+          ev.stopPropagation()
```

Comment:
why did you need this?

Reply:
I wanted to be able to support clicking the nav item to navigate when it has sublinks.

So if the arrow to collapse/expand the sublinks is clicked without this, it will expand/collapse but also navigate, which seems like bad ux.

For instance on the portfolio, the user could click the top level nav item and go to the portfolio, but they could also expand/collapse the sub items without navigating and then click a sub item to navigate.

Currently the way it is implemented on the client isn't using this and I'm just using the parent nav item to expand/collapse the sublinks as well, but I think it's nice that the component is flexible in this way

Reply:
Understood. do you mind adding a one line explanation? I find it hard to understand the reasoning behind thosw two lines without this context

Reply:
sure thing. just added it